doc/api/stream.md

```diff
@@ -1506,13 +1506,67 @@ async function print(readable) {
 print(fs.createReadStream('file')).catch(console.error);
 ```
 
-If the loop terminates with a `break` or a `throw`, the stream will be
-destroyed. In other terms, iterating over a stream will consume the stream
+If the loop terminates with a `break`, `return` or a `throw`, the stream will
+be destroyed. In other terms, iterating over a stream will consume the stream
 fully. The stream will be read in chunks of size equal to the `highWaterMark`
 option. In the code example above, data will be in a single chunk if the file
 has less then 64KB of data because no `highWaterMark` option is provided to
 [`fs.createReadStream()`][].
 
+##### `readable.iterator([options])`
```

Comment:
Just a thought: does this have to be in a new method rather than adding a parameter to the existing `Symbol.asyncIterator` one?

Reply:
I'm -0 on naming. I prefer a new method as the `Symbol.asyncIterator` one has a predefined signature by the standard.

Reply:
It doesn't feel right to me if the user has to call `[Symbol.asyncIterator]()` themselves.

Reply:
> the `Symbol.asyncIterator` one has a predefined signature by the standard.

Not really, all the standard says it this method is called with no argument (https://tc39.es/ecma262/#sec-getiterator). The standard gives a clear rule for the returned object (https://tc39.es/ecma262/#sec-asynciterable-interface), but not for the function signature. I personally don't feel strongly either way.

Reply:
I'd prefer the separate method. The key issue with extending standard-defined APIs is that it makes reasoning about the portability of code far more complicated. A separate method makes it clear. That said, the behavior of the two can be identical such that `[Symbol.asyncIterator]()` could just defer to `readable.iterator()` with default arguments.

Reply:
> That said, the behavior of the two can be identical such that `[Symbol.asyncIterator]()` could just defer to `readable.iterator()` with default arguments.

Yeah, the reason that one doesn't call the other is because of legacy streams and `this`. I'd need to use ReflectApply to bind `this`, and I preferred to have a regular method and send `this` as the first parameter instead of primordials.